src/main/java/org/apache/sling/event/impl/jobs/queues/JobExecutionContextImpl.java

```diff
@@ -84,6 +84,22 @@ public void updateProgress(final long eta) {
         }
     }
 
+    @Override
+    public void setProperty(final String name, final Object value) {
+        if ( name == null ) {
+            throw new IllegalArgumentException("Name must not be null");
```

Comment:
Optional, this could be simplfied with:

Objects.requireNonNull(name, "Name must not be null")

But this would throw a NPE (with the supplied message) rather that an IllegalArgumentMessage.

https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String-

Reply:
Since it is a user facing API, I implemented it opting for IllegalArgumentException as the exception type.  Do you agree or should I change it and allow it to be an NPE?

Reply:
We could also add a @NotNull annotation to the API to make it more visible.

Reply:
I'm okay with keeping the current code to have a more descriptive error, but I agree with @joerghoh regarding the annotations.

Reply:
I had thought of using `@NonNull` too but it meant I need to introduce a new dependency to this project and org.apache.sling.event.api project.  WDYT?
```
<dependency>
    <groupId>com.google.code.findbugs</groupId>
    <artifactId>jsr305</artifactId>
    <version>3.0.2</version>
</dependency>
```
Also, note that @Nonnull isn't perfect at blocking null parameters from getting passed, so it still make sense to keep the null checks. See here for details:
https://stackoverflow.com/questions/13484202/how-to-use-nullable-and-nonnull-annotations-more-effectively/46290602#46290602

Reply:
we use other null annotations in Sling: https://sling.apache.org/documentation/development/null-analysis.html

Reply:
Thanks @kwin, I will use those.

Reply:
Ok, done:
https://github.com/apache/sling-org-apache-sling-event/pull/26/files#diff-6821a1c0362dd00f0e0c3dd812e85e5fc3424c1b63971fe46712d547a3eafe75R89
https://github.com/apache/sling-org-apache-sling-event/pull/26/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R122-R126